app/src/main/java/io/homeassistant/companion/android/onboarding/integration/MobileAppIntegrationView.kt

```diff
@@ -97,6 +108,72 @@ fun MobileAppIntegrationView(
                     fontWeight = FontWeight.Light
                 )
             }
+            if (onboardingViewModel.deviceIsWatch && onboardingViewModel.mayRequireTlsClientCertificate) {
+                Spacer(modifier = Modifier.height(16.dp))
+
+                Text(
+                    text = stringResource(id = commonR.string.tls_cert_onboarding_title),
+                    style = MaterialTheme.typography.h6
+                )
+                Text(
+                    text = stringResource(id = commonR.string.tls_cert_onboarding_description),
+                    fontWeight = FontWeight.Light
+                )
+                Row(
+                    modifier = Modifier.fillMaxWidth(),
+                    horizontalArrangement = Arrangement.SpaceBetween
+                ) {
+                    Button(onClick = onSelectTLSCertificateClicked) {
+                        Text(text = stringResource(id = commonR.string.select_file))
+                    }
+                    Text(
+                        text = onboardingViewModel.tlsClientCertificateFilename,
+                        modifier = Modifier
+                            .align(Alignment.CenterVertically)
+                    )
+                }
+                if (onboardingViewModel.tlsClientCertificateUri != null) {
+                    Row {
```

Comment:
This `Row` is not needed as it has only one child. To make the text field use the entire screen width, use a `fillMaxWidth()` modifier instead of a `Row` + `weight(1f)` modifier.